InterfaceParser/parsers/rpc_base.py

```diff
@@ -0,0 +1,782 @@
+            param_type = Boolean(default_value=default_value)
+        elif type_name in ('Integer', 'Float', 'Float'):
+            min_value = self._extract_optional_number_attrib(
+                attrib, "minvalue", int if type_name == "Integer" else float)
+            max_value = self._extract_optional_number_attrib(
+                attrib, "maxvalue", int if type_name == "Integer" else float)
+            default_value = self._extract_optional_number_attrib(
+                attrib, "defvalue", int if type_name == "Integer" else float)
+
+            param_type = \
+                (Integer if type_name == "Integer" else Float)(
+                    min_value=min_value,
+                    max_value=max_value,
+                    default_value=default_value)
+        elif type_name == "String":
+            min_length = self._extract_optional_number_attrib(attrib, "minlength")
+            # if minlength is not defined default value is 1
+            if min_length is None:
+                min_length = 1
+            max_length = self._extract_optional_number_attrib(attrib, "maxlength")
+            default_value = self._extract_attrib(attrib, "defvalue")
+            param_type = String(min_length=min_length, max_length=max_length, default_value=default_value)
+        else:
+            if type_name.count('.') == 1:
+                custom_type_name = type_name.replace(".", "_")
+            else:
+                custom_type_name = prefix + type_name
+
+            if custom_type_name in self._types:
+                param_type = self._types[custom_type_name]
+                default_value = self._extract_attrib(attrib, "defvalue")
+                if default_value is not None:
+                    if default_value not in param_type.elements:
+                        raise ParseError("Default value '{}' for parameter '{}' is not a member of {} '{}'"
+                                         .format(default_value, params["name"], type(param_type).__name__,
+                                                 params["name"]))
+                    default_value = param_type.elements[default_value]
+            else:
+                raise ParseError("Unknown type '{}'".format(type_name))
+
+        if self._extract_optional_bool_attrib(attrib, "array", False):
+            min_size = self._extract_optional_number_attrib(attrib, "minsize")
+            max_size = self._extract_optional_number_attrib(attrib, "maxsize")
+            param_type = Array(element_type=param_type,
+                               min_size=min_size,
+                               max_size=max_size)
+
+        base_type = \
+            param_type.element_type if isinstance(param_type, Array) \
+                else param_type
+
+        other_subelements = []
+        for subelement in subelements:
+            if subelement.tag == "element":
+                if not isinstance(base_type, (Enum, EnumSubset)):
```

Comment:
Maybe this block could be placed in a new method named `_parse_base_enum_element`

Reply:
I tried to move all loop from line 597 to separate method _parse_base_enum_element, but after that  unit tests starts failing. I believe that such deep refactoring of core logic are require additional time for investigation and particular testing. As well are you sure that this change proposal right place for such refactoring?